drivers/serial/uart_npcx.c

```diff
@@ -247,7 +263,8 @@ static void uart_npcx_irq_err_disable(const struct device *dev)
 
 static int uart_npcx_irq_is_pending(const struct device *dev)
 {
-	return (uart_npcx_irq_tx_ready(dev) || uart_npcx_irq_rx_ready(dev));
+	return (uart_npcx_irq_tx_ready(dev) && uart_npcx_irq_tx_is_enabled(dev)) ||
```

Comment:
Would it make sense to handle the condition as part of `uart_npcx_irq_tx_ready` and `uart_npcx_irq_rx_ready` instead of `uart_npcx_irq_is_pending`? That way the corresponding API functions would have the same behavior.

Reply:
+1, the documentation for the `uart_irq_tx_ready()` specifies to 1 only if the interrupt is enabled and ready.

```
 * @retval 1 If TX interrupt is enabled and at least one char can be
 *           written to UART.
```

Reply:
Thanks for the suggestions. Based on the following clues I found,
1. Only `uart_irq_tx_ready()` indicates returning 0 if TX interrupt is disabled specifically. `uart_irq_rx_ready()` doesn't.
2. In uart_mcux_iuart.c, [irq_rx_ready](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/uart_mcux_iuart.c#L159) API considers the level of RX FIFO only. And in [irq_is_pending](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/uart_mcux_iuart.c#L166) API, the driver considers `irq_rx_ready()` and whether RX interrupt is enabled both. 

I prefer to move the function checks whether TX interrupt is enabled to `uart_irq_tx_ready()` only. Any comment is appreciated.

Reply:
Done.

Reply:
Interesting, other drivers have this pattern as well, the documentation for `rx` is indeed not explicit about it. Guess let's go with it.